prov/shm/src/smr.h

```diff
@@ -332,8 +332,16 @@ void smr_ep_progress(struct util_ep *util_ep);
 static inline bool smr_cma_enabled(struct smr_ep *ep,
 				   struct smr_region *peer_smr)
 {
-	return ep->region->cma_cap == SMR_CMA_CAP_ON ||
-	       ep->region == peer_smr;
+	uint8_t mask, shift;
+
+	if (ep->region == peer_smr) {
+		mask = SMR_CMA_PEER_MASK;
+		shift = SMR_CMA_SELF_SHIFT;
+	} else {
+		mask = SMR_CMA_SELF_MASK;
+		shift = SMR_CMA_PEER_SHIFT;
+	}
+	return (ep->region->cma_cap & mask) == (SMR_CMA_ON << shift);
```

Comment:
Shifting and masking just to read what should be a simple boolean value is overly complex in order to save 8-bits.  Just have 2 values that we read: peer_cma_enabled, self_cma_enabled.